src/plugins/discover/public/application/apps/main/services/use_saved_search.ts

```diff
@@ -0,0 +1,328 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import { useMemo, useEffect, useRef, useCallback } from 'react';
+import { i18n } from '@kbn/i18n';
+import { merge, Subject, BehaviorSubject } from 'rxjs';
+import { debounceTime, tap, filter } from 'rxjs/operators';
+import { isEqual } from 'lodash';
+import { DiscoverServices } from '../../../../build_services';
+import { DiscoverSearchSessionManager } from './discover_search_session';
+import {
+  IndexPattern,
+  isCompleteResponse,
+  SearchSource,
+  tabifyAggResponse,
+} from '../../../../../../data/common';
+import { SavedSearch } from '../../../../saved_searches';
+import { AppState, GetStateReturn } from './discover_state';
+import { ElasticSearchHit } from '../../../doc_views/doc_views_types';
+import { RequestAdapter } from '../../../../../../inspector/public';
+import { fetchStatuses } from '../../../components/constants';
+import { AutoRefreshDoneFn, search } from '../../../../../../data/public';
+import { calcFieldCounts } from '../utils/calc_field_counts';
+import { SEARCH_ON_PAGE_LOAD_SETTING } from '../../../../../common';
+import { validateTimeRange } from '../utils/validate_time_range';
+import { updateSearchSource } from '../utils/update_search_source';
+import { SortOrder } from '../../../../saved_searches/types';
+import { getDimensions, getChartAggConfigs } from '../utils';
+import { buildPointSeriesData, Chart } from '../components/chart/point_series';
+import { TimechartBucketInterval } from '../components/timechart_header/timechart_header';
+
+export interface UseSavedSearch {
+  refetch$: Subject<unknown>;
+  savedSearch$: SavedSearchSubject;
```

Comment:
I wonder if we could maybe rename this observable. This got me really confused the first time, since we have a `savedSearch` containing the actual savedSearch and this observable, which I thought is somewhat an observable changing when the savedSearch changes. But it seems this observable is an observable containing mostly the data (and meta data) loaded. So I wonder if it would make more sense calling it something like `data$` or `searchData$` (or some other name, I am not good at naming things :D)?

Reply:
Good points, well I think I prefer `data$` and `refetch$`, one for subscribing the data fetching changes, one to trigger those fetches. something to sleep over

Reply:
after sleeping over this, having no nightmares it should be `data$`